pegasus/helpers/section_api_helpers.rb

```diff
@@ -21,23 +21,32 @@ def self.fetch_user_students(user_id)
   def self.create(params)
     name = !params[:name].to_s.empty? ? params[:name].to_s : 'New Student'
     gender = valid_gender?(params[:gender]) ? params[:gender] : nil
+    provider = supported_provider?(params[:provider]) ? params[:provider] : 'sponsored'
     birthday = age_to_birthday(params[:age]) ?
       age_to_birthday(params[:age]) : params[:birthday]
 
     created_at = DateTime.now
 
-    row = Dashboard.db[:users].insert(
+    data =
       {
         name: name,
         user_type: 'student',
-        provider: 'sponsored',
+        provider: provider,
         gender: gender,
         birthday: birthday,
         created_at: created_at,
         updated_at: created_at,
         username: UserHelpers.generate_username(Dashboard.db[:users], name)
       }.merge(random_secrets)
-    )
+    if provider == 'sponsored'
+      row = Dashboard.db[:users].insert(data)
+    else
+      uid = params[:uid].to_s
```

Comment:
Hmm, the [documentation](http://sequel.jeremyevans.net/rdoc/files/doc/security_rdoc.html) has `to_s` littered about. My understanding (which could be incorrect) is that you are safe since you are providing the update fields as a hash rather than a string to be executed as raw SQL.